src/wallet/bdk_wallet/error.rs

```diff
@@ -0,0 +1,137 @@
+//! All Wallet-related errors.
+
+/// Enum for handling wallet-related errors.
+#[derive(Debug)]
+pub enum WalletError {
```

Comment:
Wait. Why create a new module which is a copy of the existing `WalletError`? We can just import the existing WalletError and modify it to add BDKError to it.

Reply:
The plan was to delete the old `wallet/error.rs` file after this PR. That's why I made a new `error.rs`
But will stick to your plan.